src/utils/svgFactory.ts

```diff
@@ -1,45 +1,47 @@
-import fs, { promises as pfs } from 'fs'
+import fs from 'fs'
 import { customizedConfig, flipObject } from 'common/types'
 
 class SvgFactory {
     private svgCode: string = '';
     private pathToSvg: string;
-    private outputPathToSvg: string;
     private colorCode: string;
     private rotateAngle: number;
     private flip: flipObject;
+    private customized: boolean;
 
-    constructor (pathToSvg: string, outputPathToSvg: string, customizationConfig: customizedConfig) {
+    constructor (pathToSvg: string, customizationConfig: customizedConfig, customized: boolean = false) {
       this.pathToSvg = pathToSvg
-      this.outputPathToSvg = outputPathToSvg
       this.colorCode = customizationConfig.colorCode!
       this.rotateAngle = customizationConfig.rotateAngle!
       this.flip = customizationConfig.flip!
+      this.customized = customized
     }
 
     get getSvgCode () {
       return this.svgCode
     }
 
-    readSvgCode () {
+    private readSvgCode () {
       if (fs.existsSync(this.pathToSvg)) {
         this.svgCode = fs.readFileSync(this.pathToSvg, 'utf8')
       } else {
         this.svgCode = fs.readFileSync(this.pathToSvg.replace('svg-outlined', 'svg'), 'utf8')
       }
     }
 
-    modifySvgCode () {
+    private modifySvgCode () {
       const gTagOpen = `<g ${this.changeColor()} transform="${this.rotateIcon()} ${this.flipIcon()}">`
       const index1 = this.svgCode.indexOf('>') + 1
       const index2 = this.svgCode.lastIndexOf('<')
       this.svgCode = `${this.svgCode.slice(0, index1)}${gTagOpen}${this.svgCode.slice(index1, index2)}</g></svg>`
     }
 
-    async finalizeIcon () {
+    finalizeIcon () {
       this.readSvgCode()
-      this.modifySvgCode()
-      await pfs.writeFile(this.outputPathToSvg, this.svgCode)
```

Comment:
Writing process will be moved to the controllers, when required, we would need to write an `SVG` icon probably only in the svg download API, I will make the required changes to the `ImageCustomizer` API

Reply:
ok it seems like, the `icons-customization` API, can be used to download an `svg` icons with customization also, this's can be considered as a duplication, but since we're not going to update the old routes, I will keep it as it's. 

After the full deprecation of `v1` API, we can then update this Factory.

Reply:
I'm working on a refactored version of `imageFactory` class, that would handle all cases efficiently, in fact, we don't need to create an svg files, then deleting them. `svgToPng` can take the `svg` string directly and convert it to `png`
I wil make the changes on #14